Bump blockifier to support Cairo 2.10.0

kkawula opened the pull request:
<!-- Reference any GitHub issues resolved by this PR -->

Closes #2862

## Introduced changes

<!-- A brief description of the changes -->

-

## Checklist

<!-- Make sure all of these are complete -->

- [ ] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

kkawula committed:
Update `Cargo.toml`

kkawula committed:
Fix 0

kkawula committed:
Fix imports

kkawula committed:
Undo unnecessary lock update

kkawula committed:
Use `blockifier` and `starknet_api` from crates.io

kkawula committed:
Fix 4

kkawula committed:
Rename `get_compiled_contract_class` to `get_compiled_class`

kkawula committed:
Fix imports

kkawula committed:
Update `verify_syscall_ptr` return type

kkawula committed:
Specify int size for From triat

kkawula committed:
Moc `From` impl for GasPrices and SerializableGasPrices

kkawula committed:
Fix imports

kkawula committed:
Update `get_compiled_class` return type

kkawula committed:
Revert changes in 'StarknetRuntime' struct

kkawula committed:
Add `Mock` comment

kkawula committed:
Revert `verify_syscall_ptr` type change

kkawula committed:
Add `Mock` comment

kkawula committed:
Some fixes

kkawula committed:
Add mock comment

kkawula committed:
Some fixes

kkawula committed:
Some fixes - update imports

kkawula committed:
Fix error match

kkawula committed:
Fix

cptartur committed:
Compiles

cptartur committed:
Make

cptartur committed:
Fix panic data extraction logic for string, disable for shortstrings (All blockifier tests pass)

cptartur committed:
Disable asserted gas values in non-gas tests

cptartur committed:
Move builtins test to separate file

cptartur committed:
Disable fuzzing tests

cptartur committed:
Try handling segment index offsets

cptartur committed:
Add todo

cptartur committed:
Remove todos from constants.rs

cptartur committed:
Add better comments on gas

cptartur committed:
Remove TODO in entry_code.rs

cptartur committed:
Add fixme

cptartur committed:
gas.rs wip

cptartur committed:
Restore some resources related code

cptartur committed:
Some gas changes

cptartur committed:
Add up gas from nested calls (some calculations are now correct)

cptartur committed:
Update gas tests following price changes

cptartur committed:
Add fixme to l1 gas tests

cptartur committed:
Return total of l1 gas (maintain old behavior

cptartur committed:
Set class hash for mocked calls

cptartur committed:
Restore passing used resources

cptartur committed:
Partially fix regex

cptartur committed:
Disable fuzzing gas test

cptartur committed:
Mock gas tests

cptartur committed:
Fix traces (modify blockifer code)

kkawula committed:
Merge branch 'master' into kkawula/2862-bump-cairo-and-blockifier

kkawula committed:
Fmt + typos

kkawula committed:
Fix imports

kkawula committed:
Remove MOCK remaining_gas

kkawula committed:
Implement From traint for `SerializableBlockInfo` and `GasPrices`

kkawula commented on crates/runtime/src/lib.rs:
```diff
@@ -55,39 +53,46 @@ const CAIRO_TEST_CHEATCODES: [&str; 14] = [
 
 pub struct StarknetRuntime<'a> {
     pub hint_handler: SyscallHintProcessor<'a>,
+    pub user_args: Vec<Vec<Arg>>,
```
doc.

franciszekjob committed:
Refactor returned gas vector in `calculate_used_gas`

franciszekjob committed:
Use actual `signature_length` from tx info

franciszekjob committed:
Add mocked `calculate_calldata_length`

franciszekjob committed:
Update todo

franciszekjob committed:
Pass `calldata` and `code_size`

franciszekjob committed:
Use real abi length

franciszekjob committed:
Add sierra gas calculation

franciszekjob committed:
Remove todo

franciszekjob committed:
Temporarily use 0 as abi length

franciszekjob committed:
Linting

franciszekjob committed:
Update returned values in `calculate_libfunc_cost`

kkawula commented on Cargo.toml:
```diff
@@ -25,25 +25,26 @@ license = "MIT"
 cairo-lang-macro = "0.1.0"
 cairo-vm = "1.0.0-rc3"
-cairo-annotations = "0.1.0"
+cairo-annotations = { git = "https://github.com/software-mansion/cairo-annotations.git", branch = "kkawula/2.10.0-rc.1" }
```
This dependency should be removed after `cairo-annotations` release.

kkawula replied:
https://github.com/software-mansion/cairo-annotations/pull/51

ksew1 replied:
You don't need new release, see:
- https://github.com/software-mansion/cairo-annotations/pull/51#issuecomment-2662820733
- https://github.com/foundry-rs/starknet-foundry/pull/2948

franciszekjob committed:
Refactor `calculate_sierra_gas`

franciszekjob committed:
Refactor `calculate_libfunc_cost`

franciszekjob committed:
Add comments with detailed execution resources for each gas test

franciszekjob committed:
Remove `calculate_sierra_gas` and `calculate_libfunc_cost`

THenry14 committed:
fix resources in traces

ksew1 committed:
Remove cairo-annotations branch dependency (#2948)

<!-- Reference any GitHub issues resolved by this PR -->

Related: 
-
https://github.com/foundry-rs/starknet-foundry/pull/2866#discussion_r1957835332
- https://github.com/software-mansion/cairo-annotations/pull/51
## Introduced changes

<!-- A brief description of the changes -->

- removes cairo-annotations branch dependency

## Checklist

<!-- Make sure all of these are complete -->

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

franciszekjob committed:
Remove `state_resources` param from `get_archival_data_resources`

franciszekjob committed:
Formatting

franciszekjob committed:
Fix failing tests

cptartur committed:
Fix total resources

franciszekjob committed:
Remove code size, calldata length and signature length calculations

cptartur committed:
Merge branch 'kkawula/2935-go-over-the-code-and-verify-mocks-todo-fixme' into kkawula/2862-bump-cairo-and-blockifier

franciszekjob committed:
Add comments with calculations explanations in integration gas tests

franciszekjob committed:
Remove "FIXME"; Minor comments fixes

franciszekjob committed:
Remove comment

franciszekjob committed:
Restore comments

cptartur committed:
Fix most lints

cptartur committed:
Link to copied code

cptartur committed:
Add `ENTRYPOINT_FAILED_ERROR` to `execute_inner_call` error (similar to blockifier)

cptartur committed:
Move default class hash for mocked call info logic to `mocked_call_info`

cptartur committed:
Remove commented out `resources` in `exit_nested_call`

cptartur commented on crates/forge-runner/src/build_trace_data.rs:
```diff
@@ -273,6 +272,10 @@ fn build_profiler_deprecated_syscall_selector(
+        DeprecatedSyscallSelector::GetClassHashAt => {
+            ProfilerDeprecatedSyscallSelector::GetClassHashAt
+        }
+        DeprecatedSyscallSelector::KeccakRound => unimplemented!("No corresponding syscall"),
```
@THenry14 this might be relevant for profiling

THenry14 replied:
yup :+1: should be added to cairo annotations. You can do it straight away; otherwise I will add this in a next version alongside sierra gas additions

franciszekjob committed:
Restore cast script (#2937)

<!-- Reference any GitHub issues resolved by this PR -->

Closes #2933

## Introduced changes

Restore cast script by adjusting
[`run`](https://github.com/foundry-rs/starknet-foundry/blob/ac276dde045a1304be970d75dbe91cfc1533361b/crates/sncast/src/starknet_commands/script/run.rs#L286-L408)
function

## Checklist

<!-- Make sure all of these are complete -->

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

cptartur committed:
Remove commented out code related to traces

cptartur committed:
Merge remote-tracking branch 'origin/kkawula/2862-bump-cairo-and-blockifier' into kkawula/2862-bump-cairo-and-blockifier

cptartur committed:
Allow lint in copied code

franciszekjob committed:
Add comment in `l1_handler_cost` test

franciszekjob committed:
Update asserted gas value in `multiple_storage_writes_cost` test

franciszekjob committed:
Update comment

franciszekjob committed:
Merge branch 'kkawula/2862-bump-cairo-and-blockifier' of https://github.com/foundry-rs/starknet-foundry into franciszekjob/2932-gas-changes

cptartur committed:
Remove more commented out code

franciszekjob committed:
Update comment

cptartur committed:
Use `i64` as max gas for now

cptartur committed:
Remove debug prints

cptartur committed:
Merge branch 'master' into kkawula/2862-bump-cairo-and-blockifier

# Conflicts:
#	crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs
#	crates/forge/src/run_tests/test_target.rs
#	crates/forge/tests/e2e/fuzzing.rs

cptartur committed:
Remove old fuzzing related code

cptartur committed:
Restore fuzzing tests

cptartur committed:
Bump deps

cptartur committed:
Fix typo

cptartur committed:
Closes #2954

cptartur committed:
Bump minimal scarb version

franciszekjob committed:
Merge branch 'kkawula/2862-bump-cairo-and-blockifier' of https://github.com/foundry-rs/starknet-foundry into franciszekjob/2932-gas-changes

franciszekjob committed:
Fix `multiple_storage_writes_cost` test

cptartur committed:
Link to issue regarding blockifier copied code

cptartur committed:
Link to issue regarding cairo copied code

cptartur committed:
Restore remaining fuzzing test

cptartur committed:
Remove todos

cptartur committed:
Move some constants to `runtime` crate

cptartur committed:
Use better default gas value

cptartur committed:
Remove not relevant todo

cptartur committed:
Change visibility of copied function

franciszekjob committed:
Add todo

franciszekjob committed:
Merge branch 'kkawula/2862-bump-cairo-and-blockifier' into franciszekjob/2932-gas-changes

cptartur committed:
Update cairo annotations, add missing mapping

franciszekjob committed:
Apply ode review suggestions

cptartur committed:
Use blockifier from fork

cptartur committed:
Merge branch 'kkawula/2862-bump-cairo-and-blockifier' into franciszekjob/2932-gas-changes

cptartur committed:
Link issue in code

cptartur committed:
Link issue in code

cptartur committed:
Disable default features

cptartur committed:
Merge branch 'kkawula/2862-blockifier-windows-compatible-fork' into kkawula/2862-bump-cairo-and-blockifier

cptartur committed:
Relock

cptartur committed:
Lint

cptartur committed:
Bump starknet, assert_macros, remove unused code

franciszekjob committed:
Remove `scarb_2_8_3` feature

franciszekjob committed:
Use scarb `2.8.5` in coverage and backtrace tests

franciszekjob committed:
Use scarb `2.8.5` in contracts artifacts test

franciszekjob committed:
Restore previous `MINIMAL_SCARB_VERSION`

franciszekjob committed:
Bump `assert_macros` to `2.8.2` in docs packages

franciszekjob committed:
Change starknet version in `package_matches_version_requirement_test`

franciszekjob committed:
Update starknet versions in `package_matches_version_requirement_test`

cptartur committed:
Remove `scarb 2.7.0` override in cast tests

cptartur committed:
Merge branch 'kkawula/2862-bump-cairo-and-blockifier' into 2964-docs-tests

cptartur committed:
Remove unused argument

cptartur committed:
Do not hardcode `use_kzg_da`

cptartur committed:
Link to sierra gas issue

cptartur committed:
Change fixme to todo

cptartur committed:
Copy updated code, link to source

franciszekjob committed:
Disable coverage-related tests on Windows

cptartur committed:
Merge branch '2964-docs-tests' into kkawula/2862-bump-cairo-and-blockifier

franciszekjob committed:
Trigger CI

cptartur committed:
Remove commented out code

franciszekjob committed:
Merge branch 'kkawula/2862-bump-cairo-and-blockifier' of https://github.com/foundry-rs/starknet-foundry into franciszekjob/2962-remove-cairo-2.7.0-support

franciszekjob committed:
Update warning message on outdated scarb

franciszekjob committed:
Merge branch 'master' of https://github.com/foundry-rs/starknet-foundry into kkawula/2862-bump-cairo-and-blockifier

franciszekjob committed:
Merge branch 'kkawula/2862-bump-cairo-and-blockifier' of https://github.com/foundry-rs/starknet-foundry into kkawula/2862-bump-cairo-and-blockifier

franciszekjob committed:
Merge branch 'kkawula/2862-bump-cairo-and-blockifier' of https://github.com/foundry-rs/starknet-foundry into franciszekjob/2962-remove-cairo-2.7.0-support

franciszekjob committed:
Fix clippy

franciszekjob committed:
Add `minimal_recommended_version` to `Requirement`

franciszekjob committed:
Fix failing coverage tests

franciszekjob committed:
Remove scarb version from CI step

franciszekjob committed:
Fix `test_warning_on_outdated_scarb`

franciszekjob committed:
Update changelog

franciszekjob committed:
Change default assert macros version

franciszekjob committed:
Update requirements happy case test

franciszekjob committed:
Update starknet and assert macros versions in `Scarb.toml` coverage_project

franciszekjob committed:
Use starknet `2.4.0` in `coverage_project`

franciszekjob committed:
Fix `test_fail_on_scarb_version_lt_2_8_0`

franciszekjob committed:
Update comment in `test_fail_on_scarb_version_lt_2_8_0`

franciszekjob committed:
Change test name

franciszekjob committed:
Update changelog

franciszekjob committed:
Update installation docs

franciszekjob committed:
Update changelog

kkawula committed:
Panic data regexes tests (#2940)

<!-- Reference any GitHub issues resolved by this PR -->

Closes #2931

## Introduced changes

<!-- A brief description of the changes -->

-

## Checklist

<!-- Make sure all of these are complete -->

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

franciszekjob committed:
Add code review suggestions

franciszekjob committed:
Add code review suggestions

franciszekjob committed:
Disable `coverage` mod tests instead of every test

franciszekjob committed:
Use `create_wrapper_info` instead of `create_entry_code` (#2986)

<!-- Reference any GitHub issues resolved by this PR -->

Closes #2992 

## Introduced changes

<!-- A brief description of the changes -->

-

## Checklist

<!-- Make sure all of these are complete -->

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

franciszekjob committed:
Add `warning_requirements` test

franciszekjob committed:
add tests for scarb 2.7.1

franciszekjob committed:
Fix running `warning_requirements` test on CI

franciszekjob committed:
Fix running `warning_requirements` test on CI

franciszekjob committed:
Remove `test-contracts-artifacts` job

franciszekjob committed:
Run backtrace tests on Windows

franciszekjob committed:
Revert "Run backtrace tests on Windows"

This reverts commit ce57d6f25ad688d376d09cffaaa5d8df9b762900.

franciszekjob committed:
Adjust and enable backtrace tests

franciszekjob committed:
Add `failing_requirements_on_both_minimal_versions_defined` test; Add `get_command_output` in `Requirement`

franciszekjob committed:
Fix linting

kkawula commented on crates/forge-runner/src/running.rs:
```diff
@@ -1,15 +1,28 @@
 use crate::backtrace::add_backtrace_footer;
 use crate::forge_config::{RuntimeConfig, TestRunnerConfig};
+// use crate::gas::calculate_used_gas;
```
```suggestion
```

kkawula commented on crates/forge-runner/src/running.rs:
```diff
@@ -293,14 +310,65 @@ pub fn run_test_case(
             profiling_info: None,
         }),
-        gas_used: gas,
+        // TODO return triplet
+        gas_used: u128::from(gas.l1_gas.0 + gas.l1_data_gas.0),
```
Maybe it is worth to add issue number here

franciszekjob committed:
Fix requirements logic and tests

franciszekjob committed:
Remove prints

franciszekjob committed:
Fix linting

cptartur committed:
Merge remote-tracking branch 'origin/franciszekjob/2962-remove-cairo-2.7.0-support' into kkawula/2862-bump-cairo-and-blockifier

cptartur committed:
Update crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/cairo1_execution.rs

Co-authored-by: kkawula <57270771+kkawula@users.noreply.github.com>

ksew1 committed:
Fix clean tests (#2991)

<!-- Reference any GitHub issues resolved by this PR -->

Closes #

## Introduced changes

<!-- A brief description of the changes -->

- 

## Checklist

<!-- Make sure all of these are complete -->

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

Co-authored-by: Franciszek Job <54181625+franciszekjob@users.noreply.github.com>

cptartur committed:
Link gas issue

cptartur committed:
Merge remote-tracking branch 'origin/kkawula/2862-bump-cairo-and-blockifier' into kkawula/2862-bump-cairo-and-blockifier

cptartur committed:
Update crates/forge-runner/src/running.rs

Co-authored-by: kkawula <57270771+kkawula@users.noreply.github.com>

cptartur committed:
Merge branch 'master' into kkawula/2862-bump-cairo-and-blockifier

# Conflicts:
#	Cargo.lock
#	Cargo.toml

cptartur committed:
Merge remote-tracking branch 'origin/kkawula/2862-bump-cairo-and-blockifier' into kkawula/2862-bump-cairo-and-blockifier

THenry14 commented on crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/cairo1_execution.rs:
```diff
@@ -4,60 +4,170 @@ use crate::runtime_extensions::call_to_blockifier_runtime_extension::execution::
+    write_felt(
+        &mut runner.vm,
+        &mut ptr,
+        Felt::from(0x208b_7fff_7fff_7ffe_u128),
```
do you maybe have and can provide more context on this 'magic' felt?

cptartur committed:
Merge branch 'master' into kkawula/2862-bump-cairo-and-blockifier

# Conflicts:
#	crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/cairo1_execution.rs
#	crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/calls.rs
#	crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/cheated_syscalls.rs
#	crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/deprecated/cairo0_execution.rs
#	crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/rpc.rs
#	crates/cheatnet/src/runtime_extensions/deprecated_cheatable_starknet_extension/runtime.rs
#	crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/declare.rs
#	crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/precalculate_address.rs
#	crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs
#	crates/cheatnet/src/state.rs
#	crates/cheatnet/tests/common/mod.rs
#	crates/forge-runner/src/build_trace_data.rs
#	crates/forge-runner/src/running.rs
#	crates/forge-runner/src/running/config_run.rs
#	crates/forge/tests/integration/dispatchers.rs
#	crates/runtime/src/lib.rs
#	crates/scarb-api/src/lib.rs
#	crates/sncast/src/starknet_commands/script/run.rs

cptartur commented on crates/cheatnet/src/state.rs:
```diff
@@ -207,7 +207,7 @@ impl CallTrace {
 
 /// Enum representing node of a trace of a call.
-#[derive(Clone)]
+#[derive(Clone, Debug)]
```
Added these so traces could be debugged. Was helpful when developing.

cptartur commented on crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/entry_point.rs:
```diff
@@ -289,6 +283,10 @@ fn mocked_call_info(call: CallEntryPoint, ret_data: Vec<Felt>) -> CallInfo {
         storage_read_values: vec![],
         accessed_storage_keys: HashSet::new(),
+        read_class_hash_values: vec![],
+        // This defaults to `CairoSteps` as of writing this, but it could be changed in the future
```
I'll link the new gas related issue to it

cptartur committed:
Link todo

cptartur committed:
Remove unnecessary whitespace

cptartur committed:
Update CHANGELOG.md
